CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPluginNGTest.java

```diff
@@ -0,0 +1,311 @@
+
+/**
+ *
+ * @author aldebaran30701
+ */
+public class BoxSelectionPluginNGTest {
+
+    BoxSelectionPlugin boxSelect;
+    Camera camera;
+    float[] box; // left, right, top, bottom in camera coordinates
+    float left = (float) -0.1;
+    float right = (float) 0.1;
+    float top = (float) 0.1;
+    float bottom = (float) -0.1;
+
+    private int attrX, attrY;
+    private int vxId1, vxId2, vxId3, vxId4;
+    private int txId1, txId2;
+    private int vSelectedAttrId, tSelectedAttrId;
+    private StoreGraph graph;
+
+    public BoxSelectionPluginNGTest() {
+        createNewBox();
+        camera = new Camera();
+        boxSelect = new BoxSelectionPlugin(true, true, camera, box);
+    }
+
+    private void createNewBox() {
+        box = new float[4];
+        box[0] = left;
+        box[1] = right;
+        box[2] = top;
+        box[3] = bottom;
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        graph = new StoreGraph();
+
+        VisualConcept.VertexAttribute.X.ensure(graph);
+        VisualConcept.VertexAttribute.Y.ensure(graph);
+        VisualConcept.VertexAttribute.Z.ensure(graph);
+
+        vSelectedAttrId = VisualConcept.VertexAttribute.SELECTED.ensure(graph);
+        tSelectedAttrId = VisualConcept.TransactionAttribute.SELECTED.ensure(graph);
+
+        VisualConcept.VertexAttribute.NODE_RADIUS.ensure(graph);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of edit method, of class BoxSelectionPlugin.
+     */
+    @Test
+    public void testEdit() throws InterruptedException, PluginException, IOException {
+        System.out.println("edit");
+
+        // test with 0 mix ratio
+        camera = new Camera();
+        camera.setMixRatio(0);
+        boxSelect(true, true);
+        boxSelect(true, false);
+        boxSelect(false, true);
+        boxSelect(false, false);
+
+        // Test with 1 mix ratio
+        camera = new Camera();
+        camera.setMixRatio(1);
+        boxSelect(true, true);
+        boxSelect(true, false);
+        boxSelect(false, true);
+        boxSelect(false, false);
+
+        // test with 0 mix ratio and VxVisibility
+        VisualConcept.VertexAttribute.VISIBILITY.ensure(graph);
+        camera = new Camera();
+        camera.setMixRatio(0);
+        boxSelect(true, true);
+        boxSelect(true, false);
+        boxSelect(false, true);
+        boxSelect(false, false);
+    }
+
+    /**
+     * Run a box select plugin test
+     *
+     * @param isAdd
+     * @param isToggle
+     * @throws InterruptedException
+     * @throws PluginException
+     * @throws IOException
+     */
+    public void boxSelect(final boolean isAdd, final boolean isToggle) throws InterruptedException, PluginException, IOException {
```

Comment:
In the `testEdit` method it looks like all 4 permutations of these 2 parameters are tested but I noticed that `isAdd` and `isToggle` is not used by the `boxSelect` method. Can you clarify?

Reply:
Great spotting, yeah I refactored it but forgot to add the parameters into it! Will make some changes to it - you are correct on your assumption of the `boxSelect()` method 👍

Reply:
Awesome love your work @aldebaran30701